Ajc/limit markers range of motion

ajc27-git opened the pull request:
@jonmatthis here is a first version of the Limit markers range of motion function.

The idea is to calculate the bone axes of the hand bone and then loop through a copy of the bone_definition class and calculate the local bone axes from the parent. If the local x or z axes are out of a set range then rotate the tail marker of that bone to meet the range.

Also rotate the child markers to maintain the overall posture and rigid bone state. It does a good job overall, but in certain occasions it seems that original posture was more natural. Lots of tweaks to do, adjust the rotation limits, add a factor to make it more strict or relaxed, etc.

![image](https://github.com/user-attachments/assets/366f3bb5-1840-46e0-beab-6dc456c2a1d8)

![image](https://github.com/user-attachments/assets/dfb0bff4-0b2b-4b4a-a1d2-84e25719ad2e)
![image](https://github.com/user-attachments/assets/4f022751-7a6a-4948-8839-fd5d060c3c70)

jonmatthis commented:
This is awesome! We can chat about it in the voice meeting today to see what needs to be done to get it ready to merge. 

In general, I think we should go through a lot of your recent PR's and merge them to `main` if they are ready to go. 

It won't affect the main project so long as we don't update specified version in the freemocap pyproject.toml. 

I want to start the work building the new `skellyblender` repo (and sun-setting/deprecating this one) soon, so we should get all the new functionality you've built into main quickly so it will be available to users in the addon while we build the new repo

@aaroncherian - check this out, specifically the part where he's calculating the finger bone axes in their parent's coordinate frame. We can probably adapt that method to calculate the full body angles in `skellyforge`

ajc27-git commented:
I added a general ROM scale factor in the GUI. It basically adjusts the limit_rotation_min and limit_rotation_max variables to match the factor. Factor of 1 they stay the same, between 0 and 1 it tightens the ROM and greater than 1 it loosens the ROM.

Strange that a value of 0 fixes the hands in a cupped pose. I would thought it would be a totally extended pose like the rest pose. Setting a big value like 500 brings all the ROM to [-180º,180º] resulting in no change.

I also added a selector to choose the marker to use as locked track for the hand. I put two options: index and hand_thumb_cmc. I got better results with the index marker, as we discussed it makes a more coplanar vector with the palm.

![image](https://github.com/user-attachments/assets/21626942-d3a8-4c59-ae76-7b5a64c1cd59)
